_posts/2024-03-20-strimzi-kraft-migration.md

```diff
@@ -0,0 +1,260 @@
+For this reason, there is the need to migrate existing clusters from using ZooKeeper to KRaft for storing metadata.
+We also covered the overall procedure to do that.
+The Kafka metadata migration procedure can be done without downtime, but it is not that straightforward because it requires manual configurations and several rolling updates.
+The good news is that, if your cluster is powered by Strimzi, this process is much easier, and can be driven by simply annotating your Kafka resource.
+In this blog post, we are going to show you how the Strimzi cluster operator provides a semi-automated process to run the migration easily, so that you don't need to worry about updating configurations and restarting cluster nodes.
+
+<!--more-->
+
+### Strimzi migration support
+
+If you are using the Strimzi operator to run your current ZooKeeper-based cluster on Kubernetes, the migration process is semi-automated.
+What you have to do is just updating the `strimzi.io/kraft` annotation on the `Kafka` cluster resource in order to allow the internal migration service to move across the different phases.
+The operator is going to apply the needed configuration to the nodes and roll them.
+It is also in charge of checking the migration status looking at the metrics for you to make sure it was done successfully.
+
+Before latest Strimzi 0.40.0 release, the `strimzi.io/kraft` annotation on the `Kafka` custom resource was used to define a ZooKeeper-based cluster, by using the `disabled` value, or a KRaft-based cluster, by using the `enabled` value.
+
+It is now possible to use two more values for the `strimzi.io/kraft` annotation: 
+
+* `migration` to start the migration process;
+* `rollback` to revert the migration itself;
+
+During migration or rollback, the `enabled` and `disabled` values are used to finalize the procedure by ending with a cluster in KRaft mode or keeping the ZooKeeper mode.
+
+One of the main prerequisites for the migration is about having the ZooKeeper-based cluster using the `KafkaNodePool` resource(s) to run the brokers.
+If that is not the case for you, please refer to the official Strimzi documentation [here](https://strimzi.io/docs/operators/latest/deploying#proc-migrating-clusters-node-pools-str) and this [series of blog posts](https://strimzi.io/blog/2023/08/14/kafka-node-pools-introduction/) focused on node pools.
+
+Before starting the migration, the `Kafka` custom resource has to have the `strimzi.io/node-pools: enabled` annotation to use node pools together with the `strimzi.io/kraft: disabled` which defines a ZooKeeper-based cluster.
+
+```yaml
+apiVersion: kafka.strimzi.io/v1beta2
+kind: Kafka
+metadata:
+  name: my-cluster
+  annotations:
+    strimzi.io/kraft: disabled # states that the cluster is ZooKeeper-based
+    strimzi.io/node-pools: enabled
+# ...
+```
+
+Using the Unidirectional Topic Operator and a Kafka version greater or equal 3.7.0 are important prerequisites as well.
```

Comment:
Instead of having it in the text, should there be some lists of prerequisites, so the users that would like to try it will see everything needed on one place?

Reply:
It's a blog post, usually tables are useful in documentation imho. A blog post should more tell a story.

Reply:
I didn't mean table, but just list of prerequisites:
```
Before starting the migration from ZK to KRaft, there are few prerequisites that are important to :

* use ZK-based cluster with `KafkaNodePool` for the brokers
* use Unidirectional Topic Operator
* have Kafka version greater or equal 3.7.0
```

but yeah, if you think it would not makes sense, then I'm fine with that.

Reply:
I just like the following flow because it easily allows to add links for the node pool migration and related blog posts.